src/index.ts

```diff
@@ -39,20 +39,15 @@ function addMethod(schemaType: any, name: string, fn: any) {
 
 type ObjectSchemaOf<T extends AnyObject> = ObjectSchema<
   {
-    [k in keyof T]-?: T[k] extends Array<infer E>
-      ? ArraySchema<SchemaOf<E>>
-      : T[k] extends AnyObject
-      ? // we can't use  ObjectSchema<{ []: SchemaOf<T[k]> }> b/c TS produces a union of two schema
-        ObjectSchemaOf<T[k]>
-      : BaseSchema<Maybe<T[k]>, AnyObject, T[k]>;
+    [k in keyof T]-?: SchemaOf<T[k]>;
```

Comment:
Regarding the the nested-lazy-inside-object concern, how about this? That doesn't cause any compile issues in my codebase, so I think it's valid.
```suggestion
    [k in keyof T]-?: SchemaOf<T[k]> | Lazy<SchemaOf<T[k]>>;
```

Reply:
we can try it!

Reply:
And doing it like this still avoids putting `Lazy` in the top level of `SchemaOf` so it's still safe in that regard too.